### PR TITLE
Port kilosort motion estimation

### PR DESCRIPTION
**Pull request by alejoe91**
Port Kilosort drift estimation from pykilsort:
https://github.com/int-brain-lab/pykilosort/blob/ibl_prod/pykilosort/datashift2.py#L537

Differences:
- kilosort has its own spike detection, SI uses peaks and peak locations detected with sorting components
- inputs are unified to SI API
- amplitude histogram normalization is different, since SI uses true amplitudes, while KS uses amplitude scaling from [`standalone_detector`](https://github.com/int-brain-lab/pykilosort/blob/ibl_prod/pykilosort/datashift2.py#L428) (which uses template-matching with generic templates)
- renamed most variables to improve readibility

Here is a gist which shows the output and the differences in implementations:
https://gist.github.com/alejoe91/67f2aa6d544d3ecb1d516f34f54bba48

### CODE REVIEW

File: spikeinterface/sortingcomponents/motion_estimation.py [hunk 1]
```diff
@@ -180,9 +195,89 @@ def estimate_motion(recording, peaks, peak_locations,
...
+        chunk_size = ensure_chunk_size(recording, chunk_duration=f"{bin_duration_s}s")
+        chunks = divide_recording_into_chunks(recording, chunk_size)
+        n_amp_bins = method_kwargs['n_amp_bins']
+
+        # min and max for the raspike_locsepths
+        dmin = min(contact_pos) - 1
+        dmax = max(contact_pos)
+        num_spatial_bins = len(spatial_bins) - 1
+        # num_spatial_bins = int(1 + np.ceil(np.max(contact_pos) - dmin) / bin_um)
+        spike_depths = peak_locations["y"]
```
Comment (samuelgarcia) [inline spikeinterface/sortingcomponents/motion_estimation.py:213]:
```suggestion
        spike_depths = peak_locations[direction]
```

**Comment by alejoe91** [comment]
@samuelgarcia @yger @bendichter 

Ready to review on my side. Check out this gist too: https://gist.github.com/alejoe91/f4037c8c7466739667fb7f4f9014316f

File: spikeinterface/sortingcomponents/motion_estimation.py [hunk 2]
```diff
@@ -180,9 +197,45 @@ def estimate_motion(recording, peaks, peak_locations,
                 progress_bar=False,
             )
             motion.append(one_motion[:, np.newaxis])
-
         motion = np.concatenate(motion, axis=1)
 
+    elif method == "kilosort25": # TODO: change name to: "iterative_histogram_registration"
```
Comment (samuelgarcia) [inline spikeinterface/sortingcomponents/motion_estimation.py:202]: Lets merge without "kilosort25" but "iterative_histogram_registration" or maybe "iterative_template_registration" no ?

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
really cool.

**Pull request merged**